p2p/src/peer_manager/peerdb/mod.rs

```diff
@@ -90,11 +93,16 @@ pub struct PeerDb<T: NetworkingService> {
     ///
     /// The duration represents the `UNIX_EPOCH + duration` time point, so the ban should end
     /// when `current_time > ban_duration`.
-    banned: BTreeMap<T::BannableAddress, Duration>,
+    banned_addresses: BTreeMap<T::BannableAddress, Duration>,
+
+    store: Box<dyn PeerDbStore>,
 }
 
 impl<T: NetworkingService> PeerDb<T> {
-    pub fn new(p2p_config: Arc<config::P2pConfig>) -> crate::Result<Self> {
+    pub fn new(
+        p2p_config: Arc<config::P2pConfig>,
+        store: Box<dyn PeerDbStore>,
```

Comment:
Why use dyn when you can use generics?

Reply:
Is it OK now? I still think it was better with `dyn`.